server/controllers/navigation.js

```diff
@@ -19,10 +19,25 @@ const errorHandler = (ctx) => (error) => {
   throw error;
 };
 
-module.exports = {
+module.exports = ({strapi}) => ({
   async config() {
     return getService().config();
   },
+
+  async updateConfig(ctx) {
+    await getService().updateConfig(ctx.request.body)
+    if (strapi.plugin('graphql'))
+      setImmediate(() => strapi.reload());
```

Comment:
Extract this to separated method, worth to make is easy to understand for user what is happening